src/report/lint-report.ts

```diff
@@ -0,0 +1,213 @@
+  }
+
+  core.warning(
+    `Lint report summary exceeded limit of ${MAX_REPORT_LENGTH} bytes and will be trimmed`
+  )
+  return trimReport(lines)
+}
+
+function trimReport(lines: string[]): string {
+  const closingBlock = '```'
+  const errorMsg = `**Report exceeded GitHub limit of ${MAX_REPORT_LENGTH} bytes and has been trimmed**`
+  const maxErrorMsgLength = closingBlock.length + errorMsg.length + 2
+  const maxReportLength = MAX_REPORT_LENGTH - maxErrorMsgLength
+
+  let reportLength = 0
+  let codeBlock = false
+  let endLineIndex = 0
+  for (endLineIndex = 0; endLineIndex < lines.length; endLineIndex++) {
+    const line = lines[endLineIndex]
+    const lineLength = getByteLength(line)
+
+    reportLength += lineLength + 1
+    if (reportLength > maxReportLength) {
+      break
+    }
+
+    if (line === '```') {
+      codeBlock = !codeBlock
+    }
+  }
+
+  const reportLines = lines.slice(0, endLineIndex)
+  if (codeBlock) {
+    reportLines.push('```')
+  }
+  reportLines.push(errorMsg)
+  return reportLines.join('\n')
+}
+
+function renderLintReport(lintIssues: LintIssue[], baseUrl: string): string[] {
+  const sections: string[] = []
+
+  sections.push('# Android Lint Results\n\n')
+
+  const badges = getLintReportBadges(lintIssues)
+  sections.push(...badges)
+
+  const issues = getLintIssuesReport(lintIssues, baseUrl)
+  sections.push(...issues)
+
+  return sections
+}
+
+function getLintIssuesReport(lintIssues: LintIssue[], baseUrl: string): string[] {
+  const sections: string[] = []
+
+  sections.push('## Summary\n\n')
+
+  if (lintIssues.length > 1) {
+    const categories = [...new Set(lintIssues.map(li => li.category))].map((cat, catIndex) => {
+        const category = cat
+        const index = catIndex
+        return {category, index}
+    })
+    const issueDetails: string[] = []
+    for (const cat of categories) {
+      const idTables: string[] = []
+      sections.push(`### ${cat.category}`)
+      const categoryData = lintIssues.filter(li => li.category === cat.category)
+      const ids = [...new Set(categoryData.map(li => li.id))].map((li, idIndex) => {
+          const issueId = li
+          const index = idIndex
+          return {issueId, index}
+      })
+      const categorySummaryRows: string[][] = []
+      for (const id of ids) {
+        const idData = categoryData.find(cd => cd.id === id.issueId)
+        const idRows = categoryData.filter(cd => cd.id === id.issueId)
+        const count = idRows.length
+        if (idData && idRows && count > 0) {
+          const lintSlug = makeLintIssueSlug(cat.index, id.index)
+          const addr = baseUrl + lintSlug.link
+          const headerLink = link(id.issueId, addr)
+          categorySummaryRows.push([
+            count.toString(),
+            headerLink,
+            idData.summary,
+            `${getSeverityIcon(idData)}`
+          ])
+          const nameLink = `<a id="${lintSlug.id}" href="${baseUrl + lintSlug.link}">${idData.summary}</a>`
```

Comment:
🤷‍♂️  I thought it was nice to have in the markdown, as it has a more detailed explanation than just the "summary" text you normally see, but to each his own. I was basically following the format of the lint html file that gets generated when running lint (more or less). I suppose it could be an optional thing.